tests/mutation_test.cpp

```diff
@@ -563,70 +568,129 @@ TEST_CASE( "All_valid_mutations_can_be_purified", "[mutations][purifier]" )
     }
 }
 
-//The chance of a mutation being bad is a function of instability, see
-//Character::roll_bad_mutation. This can't be easily tested on in-game
-//mutations, because exceptions exist - e.g., you can roll a good mutation, but
-//end up mutating a bad one, because the bad mutation is a prerequisite for the one
-//you actually rolled.
-//For this reason, this tests on an artificial category that doesn't
-//mix good and bad mutations within trees. Additionally, it doesn't contain
-//neutral mutations, because those are always allowed.
-//This also incidentally tests that, given available mutations and enough mutagen,
-//Character::mutate always succeeds to give exactly one mutation.
+// your odds of getting a good or bad mutation depend on what mutations you have and what tree you're in.
+// you are given an integer that is first multiplied by 0.5 and then divided by the total non-bad mutations in the tree.
+// for example if you mutate into Alpha, that tree has 21 mutations (as of 3/14/2024) that aren't bad.
+// Troglobite has 28 mutations that aren't bad. (as of 3/4/2024).
+// Thus for all things equal, Alpha gets more bad mutations per good mutation than Trog,
+// but since Trog gets more good muts total it has more chances to get bad mutations.
+// The aforementioned "integer" is increased from 0 by 1 for each non-bad mutation you have.
+// mutations you have are counted as two if they don't belong to the tree you're mutating into.
+// Starting traits are never counted, and bad mutations are never counted. Only "valid" (mutable) mutations count.
+// This test case compares increasing instability in both Alpha and Trog as more mutations are added.
+// Given that Alpha and Trog share some mutations but not all, they should either increase instability at the same or different rates.
+// This test then also checks and makes sure that Alpha gets more bad mutations than Trog given the same total instability in each.
+// After that it checks to make sure each tree gets more bad mutations than before after increasing effective instability by 4x.
```

Comment:
```suggestion
// Your odds of getting a good or bad mutation depend on what mutations you have and what tree you're in.
// You are given an integer that is first multiplied by 0.5 and then divided by the total non-bad mutations in the tree.
// For example if you mutate into Alpha, that tree has 21 mutations (as of 3/14/2024) that aren't bad.
// Troglobite has 28 mutations that aren't bad. (as of 3/4/2024).
// Thus, for all things equal, Alpha gets more bad mutations per good mutation than Trog,
// but since Trog gets more good muts total it has more chances to get bad mutations.
// The aforementioned "integer" is increased from 0 by 1 for each non-bad mutation you have.
// Mutations you have are counted as two if they don't belong to the tree you're mutating into.
// Starting traits are never counted, and bad mutations are never counted. Only "valid" (mutable) mutations count.
// This test case compares increasing instability in both Alpha and Trog as more mutations are added.
// Given that Alpha and Trog share some mutations but not all, they should either increase instability at the same or different rates.
// This test then also checks and makes sure that Alpha gets more bad mutations than Trog given the same total instability in each.
// After that it checks to make sure each tree gets more bad mutations than before after increasing effective instability by 4x.
```